### PR TITLE
Dont allow entry to nil map

### PR DESCRIPTION
**Pull request by cbron**

### CODE REVIEW

**Review by daxmc99 — APPROVED** [review]

File: mappers/setvalue.go [hunk 1]
```diff
@@ -11,13 +11,13 @@ type SetValue struct {
 }
 
 func (d SetValue) FromInternal(data map[string]interface{}) {
-	if d.ExternalValue != nil {
+	if len(data) != 0 && d.ExternalValue != nil {
```
Comment (StrongMonkey) [inline mappers/setvalue.go]: Can we do nil check? non-nil map can have zero len.

Comment (cbron) [reply mappers/setvalue.go]: fixed

**Review by StrongMonkey — APPROVED** [review]

**Comment by ibuildthecloud** [comment]
This change needs to be done on rancher/norman v2 branch.  This code isn't used anymore.

**Comment by cbron** [comment]
https://github.com/rancher/rio/issues/718

**Pull request closed without merging**